### PR TITLE
Feature/checkpoint

### PR DESCRIPTION
**Pull request by RickyVimon**
What's new?

- CheckPoints system with 3 lives, respawning at the closest passed checkpoint when the player dies or at the beggining otherwise.

- New UI indicator on the HUD showing the Remaining lives.

- Camera bug fixed when teleporting to zone 0.

- RestartEvents()

### CODE REVIEW

File: Scripts/src/Script/UILives.cpp [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+// 	config.AddUInt(example->UUID, "ExampleNameforSave");
+// 	Script::Save(config);
+// }
+
+// //Use this for linking GO AND VARIABLES automatically
+// void UILives::Load(const Config& config)
+// {
+// 	exampleUUID = config.GetUInt("ExampleNameforSave", 0);
+// 	Script::Load(config);
+// }
```
Comment (JorxSS) [inline Scripts/src/Script/UILives.cpp:143]: delete unused code

File: Scripts/src/Script/UILives.h [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+	void DecreaseLives();
+	void DeactiveAll();
+
+
+	void OnInspector(ImGuiContext*) override;
+	void InitPublicGameObjects();
+
+	//void Save(Config& config) const override;
+	//void Load(const Config& config) override;
```
Comment (JorxSS) [inline Scripts/src/Script/UILives.h:30]: delete unused code

File: Scripts/src/Script/CheckPointManager.cpp [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+	{
+		new_distance = (checkpoints[i]->zone_to_spawn + 1) - current_zone;
+		if (new_distance >= 0 && new_distance < shortest_distance)
+		{
+			closest_checkpoint = checkpoints[i];
+			shortest_distance = new_distance;
+		}
+	}
+
+	return closest_checkpoint;
```
Comment (Dartemiss) [inline Scripts/src/Script/CheckPointManager.cpp:117]: This  can be const i think

File: Scripts/src/Script/EventManager.cpp [hunk 4]
```diff
@@ -520,29 +579,16 @@ void EventManager::GetZones()
...
-
+		new_zone->dissolver->RestartDisolve();
 		zone_list.emplace_back(new_zone);
 	}
 }
 
+int EventManager::GetCurrentZone()
+{
+	return current_zone;
+}
```
Comment (Dartemiss) [inline Scripts/src/Script/EventManager.cpp:590]: const

File: Scripts/src/Script/UILives.cpp [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+	}
+}
+
+void UILives::DeactiveAll()
+{
+	for (size_t i = 0; i < lives_indicator.size(); i++)
+	{
+		lives_indicator[i]->Disable();
+	}
+}
```
Comment (Dartemiss) [inline Scripts/src/Script/UILives.cpp:100]: I think this can be const

**Review by Dartemiss — APPROVED** [review]
Very good job man! We found a minor bug where when you spawn when dying on zone 3 you listen the roots sound.

Just check the code tweaks that I commented and go for it :D

**Review by JorxSS — APPROVED** [review]

**Pull request merged**